src/app/services/cache/cache.service.ts

```diff
@@ -185,63 +197,31 @@ export class CacheService implements OnDestroy {
   }
 
   getSystemDetails(): Subject<SystemDetails> {
-    if (!isDefined(this.systemDetailsSubscription)) {
+    if (!isDefined(this.systemDetailsSubject)) {
       this.systemDetailsSubject = new Subject<SystemDetails>();
-      this.messageService.getSystemDetails().then((systemDetails: SystemDetails) => this.systemDetailsCache = systemDetails);
-
-      this.systemDetailsTimer = Observable.interval(CacheService.SYSTEM_DETAILS_POLLING_RATE);
-      this.systemDetailsSubscription = this.systemDetailsTimer.subscribe(() => {
-        this.messageService.getSystemDetails().then((systemDetails: SystemDetails) => {
-          // todo: check to see if any updates have occurred, only emit if there is a change
-          this.systemDetailsCache = systemDetails;
-          this.systemDetailsSubject.next(this.systemDetailsCache);
-        });
-      });
+      this.fetchSystemDetailsRecursive();
       return this.systemDetailsSubject;
     } else {
       setTimeout(() => this.systemDetailsSubject.next(this.systemDetailsCache));
```

Comment:
Am i correct in assuming the purpose of this timeout is to make sure that whatever called this function has gotten the return back before the next is called?